dotcom-rendering/src/web/components/NewslettersList.tsx

```diff
@@ -0,0 +1,151 @@
+import { css } from '@emotion/react';
+import {
+	brandAlt,
+	headline,
+	neutral,
+	space,
+} from '@guardian/source-foundations';
+import { LinkButton } from '@guardian/source-react-components';
+import type { EditionLinkType } from '../../model/extract-nav';
+import type { Newsletter } from '../../types/content';
+import type { EditionId } from '../lib/edition';
+import { getEditionFromId } from '../lib/edition';
+import { Hide } from './Hide';
+import { NewsletterPrivacyMessage } from './NewsletterPrivacyMessage';
+import { Section } from './Section';
+
+export interface NewslettersListProps {
+	newsletters: Newsletter[];
+	headingText: string;
+	mmaUrl?: string;
+	editionId: EditionId;
+}
+
+type GroupedNewsletters = { groupName: string; newsletters: Newsletter[] }[];
+
+const putNewslettersInGroups = (
```

Comment:
In DCR we try to avoid doing much data modification / enhancement in the components, preferring instead the 'enhancement' (transformation really) model we use for Fronts.